### ISSUE TITLE
Update `init` behavior

### ISSUE THREAD

**Issue by jamesplease** [conorhastings/use-reducer-with-side-effects#9]
When using `useReducer`, init returns the state:

```js
useReducer(reducer, initialState, () => {
  return {
    someState: true
  };
})
```

With the current API of this library, you must return the "internal" structure of the store:

```js
useReducerWithSideEffects(reducer, initialState, () => {
  return {
    state: { someState: true },
    sideEffects: [] // you also must include this or else an uncaught exception will occur
  };
})
```

Users are unable to use `Update`/`UpdateWithSideEffects`/etc., because those don't return the correct key: they return `newState` rather than `state`.

I suggest refactoring `Update`/etc. to internally use the key `state` instead. That way, users of `init` can use a consistent API with this lib:

```js
useReducerWithSideEffects(reducer, initialState, () => {
  return Update({ someState: true });
})
```

This change could even be released as a bug fix, I think, since it only changes non-public structures.

I'm curious to hear your thoughts on this @conorhastings !

**Comment by conorhastings** [comment]
haven't tested but esseentially something like this? https://github.com/conorhastings/use-reducer-with-side-effects/pull/11

**Comment by jamesplease** [comment]
Resolved in 2.0.0-beta.2
